certbot/client.py

```diff
@@ -326,10 +318,20 @@ def obtain_certificate(self, domains):
                 self.config.rsa_key_size, self.config.key_dir)
             csr = crypto_util.init_save_csr(key, domains, self.config.csr_dir)
 
-        certr, chain = self.obtain_certificate_from_csr(
-            domains, csr, authzr=authzr)
+        orderr = self.acme.new_order(csr.data)
+        authzr = self.auth_handler.handle_authorizations(orderr, self.config.allow_subset_of_names)
+        auth_domains = set(a.body.identifier.value for a in authzr)
+        successful_domains = [d for d in domains if d in auth_domains]
+
+        if successful_domains != domains:
```

Comment:
I don't think you can compare set and list objects like this.

Reply:
I made the same mistake when testing with ipdb.